src/pkg/services/m365/api/graph/http_wrapper.go

```diff
@@ -120,13 +120,17 @@ func (hw httpWrapper) Request(
 			break
 		}
 
-		if IsErrApplicationThrottled(err) {
-			return nil, Stack(ctx, clues.Stack(core.ErrApplicationThrottled, err))
+		err = stackWithCoreErr(ctx, err, 1)
```

Comment:
hmmm I'm trying to figure out if a slight rearrangement of the code would be cleaner here. I'm not sure, but maybe it would help since we could have just a single if-block to check for errors? I was thinking of something like the following for the loop body:
1. make request
2. unconditionally call `stackWithCoreErr`
3. if the error is an http2StreamError: sleep and use `continue` to try again
4. otherwise: break out of the loop

Reply:
I think that's largely what we're doing, just a twist on how we continue/break.  This should also get cleaned up a little in the next PR.  I needed to deduplicate the error stacking already.